packages/fx-core/src/core/environment.ts

```diff
@@ -82,9 +86,28 @@ class EnvironmentManager {
     }
 
     envName = envName ?? this.getDefaultEnvName();
-    const configResult = await this.loadEnvConfig(projectPath, envName);
+    let configResult = await this.loadEnvConfig(projectPath, envName);
     if (configResult.isErr()) {
-      return err(configResult.error);
+      if (envName === "local") {
```

Comment:
Maybe only generate the local config file when it doesn't exist, or user customized configs will be lost.

Reply:
Move to check local config file location.